source/common/stats/thread_local_store.h

```diff
@@ -16,6 +16,86 @@
 namespace Envoy {
 namespace Stats {
 
+/**
+ * Log Linear Histogram implementation per thread.
+ */
+class ThreadLocalHistogramImpl : public Histogram, public MetricImpl {
+public:
+  ThreadLocalHistogramImpl(const std::string& name, Store& parent, std::string&& tag_extracted_name,
+                           std::vector<Tag>&& tags);
+
+  virtual ~ThreadLocalHistogramImpl();
+  // Stats::Histogram
+  void recordValue(uint64_t value) override;
+
+  // TODO(ramaraochavali): split the Histogram interface in to two - parent and tls.
```

Comment:
let's just do it now per above

Reply:
ok.just trying to avoid review churn. But will do it.